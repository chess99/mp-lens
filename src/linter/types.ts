/**
 * Types for the linting feature
 */

/**
 * Represents an issue found during linting
 */
export interface LintIssue {
  /** Unique id for the page/component (e.g. page:pages/xxx or comp:components/xxx) */
  id: string;
  /** Path to the WXML file */
  wxmlFile: string;
  /** Path to the JSON file */
  jsonFile: string;
  /** Components declared in JSON but not used in WXML */
  declaredNotUsed: Array<{
    /** Component tag name */
    componentTag: string;
    /** Path to the component (from usingComponents) */
    path?: string;
  }>;
  /** Tags used in WXML but not declared in JSON */
  usedNotDeclared: Array<{
    /** Component tag name */
    componentTag: string;
    /** WXML files where this tag was used */
    usedIn: string[];
    /** Suggestion for fixing the issue */
    suggestion?: string;
  }>;
}

/**
 * Represents the overall result of linting
 */
export interface LintResult {
  /** Summary statistics */
  summary: {
    /** Total number of files scanned */
    filesScanned: number;
    /** Number of files with issues */
    filesWithIssues: number;
    /** Total count of declared but not used components */
    declaredNotUsedCount: number;
    /** Total count of used but not declared components */
    usedNotDeclaredCount: number;
  };
  /** List of issues found */
  issues: LintIssue[];
}
