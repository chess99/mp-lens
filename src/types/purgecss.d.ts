declare module 'purgecss' {
  interface UserDefinedSafelist {
    standard?: (string | RegExp)[];
    deep?: RegExp[];
    greedy?: RegExp[];
    variables?: (string | RegExp)[];
    keyframes?: (string | RegExp)[];
  }

  interface PurgeCSSUserDefinedOptions {
    content: (string | { raw: string; extension: string })[];
    css: (string | { raw: string; name?: string })[];
    defaultExtractor?: (content: string) => string[];
    extractors?: { extractor: any; extensions: string[] }[];
    fontFace?: boolean;
    keyframes?: boolean;
    rejected?: boolean;
    rejectedCss?: boolean;
    safelist?: (string | RegExp)[] | UserDefinedSafelist;
    variables?: boolean;
    // stdin?: boolean; // Less common for programmatic use
    // stdout?: boolean; // Less common for programmatic use
    // config?: string; // For config file, not direct options
    // output?: string; // For CLI output, not programmatic return
  }

  interface PurgeCSSResult {
    css: string;
    file?: string;
    rejected?: string[];
    rejectedCss?: string;
  }

  export class PurgeCSS {
    constructor(); // In PurgeCSS v5, constructor typically does not take options.
    purge(options: PurgeCSSUserDefinedOptions): Promise<PurgeCSSResult[]>;
  }
}
