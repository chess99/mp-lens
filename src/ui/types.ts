// 树状图数据类型
export interface TreeNodeData {
  id: string;
  label: string;
  type: string;
  properties?: {
    fileCount?: number;
    totalSize?: number;
    fileTypes?: Record<string, number>;
    sizeByType?: Record<string, number>;
    [key: string]: any;
  };
  children?: TreeNodeData[];
  parent?: string;
  collapsed?: boolean;
}

// 图表类型
export interface ChartData {
  labels: string[];
  values: number[];
  colors?: string[];
}

// 统计数据类型
interface Statistics {
  totalFiles: number;
  totalCodeSize: number;
  totalPages: number;
  totalComponents: number;
  fileTypes: Record<string, number>;
  sizeByType: Record<string, number>;
}

// 应用程序Props类型
export interface AppProps {
  data: TreeNodeData;
  statistics?: Statistics;
}

// 树状图组件Props类型
export interface TreeViewProps {
  data: TreeNodeData;
  onNodeSelect: (node: TreeNodeData) => void;
  selectedNodeId?: string;
  onExpandAll?: () => void;
  onCollapseAll?: () => void;
}

// 节点详情Props类型
export interface NodeDetailsProps {
  node: TreeNodeData;
}

// 图表组件Props类型
interface ChartProps {
  title: string;
  data: ChartData;
  type: 'pie' | 'bar';
  isBytes?: boolean;
}

// Tab组件Props类型
export interface TabsProps {
  tabs: {
    id: string;
    label: string;
    content: any;
  }[];
}

// 全局window接口扩展
declare global {
  interface Window {
    __TREE_DATA__: TreeNodeData;
    __STATISTICS__: Statistics;
    preactApp: {
      hydrate: () => void;
    };
  }
}
