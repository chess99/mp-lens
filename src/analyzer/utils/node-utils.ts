import * as path from 'path';

/**
 * 生成 Page/Component 节点的 id 和 label
 * @param type 'Page' | 'Component'
 * @param basePath 绝对路径
 * @param miniappRoot miniapp 根目录
 */
export function getNodeIdAndLabel(
  type: 'Page' | 'Component',
  basePath: string,
  miniappRoot: string,
): { id: string; label: string } {
  const rel = path.relative(miniappRoot, basePath).replace(/\\/g, '/');
  return {
    id: type === 'Page' ? `page:${rel}` : `comp:${rel}`,
    label: rel,
  };
}
