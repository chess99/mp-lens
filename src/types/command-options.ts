export interface CommandOptions {
  project: string;
  verbose?: boolean;
  verboseLevel?: number;
  config?: string;
}

export interface AnalyzerOptions {
  fileTypes: string[];
  excludePatterns?: string[];
  verbose?: boolean;
  verboseLevel?: number;
  essentialFiles?: string[];
  miniappRoot?: string;
  entryFile?: string;
  entryContent?: any;
  keepAssets?: string[];
}

export interface OutputOptions {
  format: 'text' | 'json';
  projectRoot: string;
  miniappRoot?: string;
}

export interface GraphOptions extends CommandOptions {
  format: 'html' | 'dot' | 'json' | 'png' | 'svg';
  output?: string;
  depth?: number;
  focus?: string;
  npm?: boolean;
  miniappRoot?: string;
  entryFile?: string;
}

export interface CleanOptions extends CommandOptions {
  types: string;
  exclude: string[];
  essentialFiles?: string;
  dryRun: boolean;
  backup?: string;
  yes: boolean;
  miniappRoot?: string;
  entryFile?: string;
}

export interface ListUnusedOptions extends CommandOptions {
  types: string;
  exclude: string[];
  essentialFiles?: string;
  outputFormat: 'text' | 'json';
  output?: string;
  miniappRoot?: string;
  entryFile?: string;
}

/**
 * 配置文件中可用的选项
 * 包含从配置文件加载的所有可能选项
 */
export interface ConfigFileOptions {
  // 基本选项
  miniappRoot?: string; // 小程序代码子目录
  entryFile?: string; // 入口文件路径

  // 文件分析相关
  types?: string; // 要检查的文件类型
  exclude?: string[]; // 要排除的文件/目录
  excludePatterns?: string[]; // 要排除的文件/目录（兼容旧的API）
  essentialFiles?: string[] | string; // 必要文件列表
  keepAssets?: string[]; // Patterns for assets to always keep

  // 别名相关
  aliases?: {
    // 别名配置
    [key: string]: string | string[];
  };
  aliasMap?: Record<string, string>; // 别名映射（简化版）

  // 输出相关
  outputFormat?: 'text' | 'json'; // list-unused命令输出格式

  // 图表生成相关
  format?: 'html' | 'dot' | 'json' | 'png' | 'svg'; // graph命令输出格式
  graphFormat?: 'html' | 'dot' | 'json' | 'png' | 'svg'; // graph命令输出格式（替代名称）
  depth?: number; // 图表依赖深度限制
  graphDepth?: number; // 图表依赖深度限制（替代名称）
  includeNpm?: boolean; // 是否包含npm依赖
  npm?: boolean; // 是否包含npm依赖（替代名称）
  focus?: string; // 要关注的特定文件

  // 清理相关
  dryRun?: boolean; // 是否仅模拟删除
  backup?: string; // 备份目录路径
  backupDir?: string; // 备份目录路径（替代名称）
  yes?: boolean; // 是否跳过确认

  // 高级选项
  entryContent?: any; // 自定义入口内容

  // 输出路径
  output?: string; // 输出文件路径
}
