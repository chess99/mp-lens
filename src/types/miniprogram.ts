/**
 * 微信小程序 app.json 相关类型定义
 */

/**
 * 页面窗口表现配置
 */
export interface MiniProgramWindowConfig {
  navigationBarBackgroundColor?: string;
  navigationBarTextStyle?: 'black' | 'white';
  navigationBarTitleText?: string;
  navigationStyle?: 'default' | 'custom';
  backgroundColor?: string;
  backgroundTextStyle?: 'light' | 'dark';
  backgroundColorTop?: string;
  backgroundColorBottom?: string;
  enablePullDownRefresh?: boolean;
  onReachBottomDistance?: number;
  pageOrientation?: 'auto' | 'portrait' | 'landscape';
  restartStrategy?: 'homePage' | 'homePageAndLatestPage';
  initialRenderingCache?: 'static' | 'dynamic';
  visualEffectInBackground?: 'hidden' | 'none';
  handleWebviewPreload?: 'static' | 'none';
  [key: string]: any; // 支持自定义字段
}

/**
 * TabBar 配置项
 */
export interface MiniProgramTabBarItem {
  pagePath: string;
  text: string;
  iconPath?: string;
  selectedIconPath?: string;
}

/**
 * TabBar 配置
 */
export interface MiniProgramTabBar {
  color?: string;
  selectedColor?: string;
  backgroundColor?: string;
  borderStyle?: 'black' | 'white';
  position?: 'bottom' | 'top';
  custom?: boolean;
  list: MiniProgramTabBarItem[];
}

/**
 * 网络超时配置
 */
export interface MiniProgramNetworkTimeout {
  request?: number;
  connectSocket?: number;
  uploadFile?: number;
  downloadFile?: number;
}

/**
 * 权限配置
 */
export interface MiniProgramPermission {
  scope: string;
  desc: string;
}

/**
 * 用于声明式组件
 */
export interface MiniProgramUsingComponents {
  [componentName: string]: string;
}

/**
 * Worker 配置
 */
export interface MiniProgramWorkers {
  workers: string;
}

/**
 * 分包配置
 */
export interface MiniProgramSubPackage {
  root: string;
  name?: string;
  pages: string[];
  independent?: boolean;
}

/**
 * 插件配置
 */
export interface MiniProgramPlugins {
  [pluginName: string]: {
    version: string;
    provider: string;
  };
}

/**
 * 预加载规则
 */
export interface MiniProgramPreloadRule {
  [path: string]: {
    network?: 'all' | 'wifi';
    packages: string[];
  };
}

/**
 * 小程序 app.json 配置
 */
export interface MiniProgramAppJson {
  pages: string[];
  subPackages?: MiniProgramSubPackage[];
  subpackages?: MiniProgramSubPackage[]; // 兼容 subpackages 拼写
  window?: MiniProgramWindowConfig;
  tabBar?: MiniProgramTabBar;
  networkTimeout?: MiniProgramNetworkTimeout;
  debug?: boolean;
  functionalPages?: boolean;
  plugins?: MiniProgramPlugins;
  preloadRule?: MiniProgramPreloadRule;
  resizable?: boolean;
  usingComponents?: MiniProgramUsingComponents;
  permission?: { [scope: string]: { desc: string } };
  sitemapLocation?: string;
  style?: string;
  useExtendedLib?: { [libName: string]: boolean };
  entranceDeclare?: { locationId: number; serviceName: string };
  darkmode?: boolean;
  themeLocation?: string;
  lazyCodeLoading?: 'requiredComponents';
  singlePage?: { navigationBarFit?: 'float' };
  supportedMaterials?: { materialType: string }[];
  serviceProviderTicket?: string;
  workers?: string;
  requiredBackgroundModes?: string[];
  requiredPrivateInfos?: string[];
  visualEffectInBackground?: string;
  renderer?: 'webview' | 'skyline';
  rendererOptions?: { skyline?: { defaultDisplayBlock?: boolean } };
  [key: string]: any; // 支持自定义字段
}
