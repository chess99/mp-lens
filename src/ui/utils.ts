// 与 analyzer fileCount 逻辑一致的工具函数

import type { GraphLink, GraphNode } from '../analyzer/project-structure';

/**
 * 统计结构节点（如 App/Page/Package/Component）下所有可达 Module 节点
 * @param nodes ProjectStructure.nodes
 * @param links ProjectStructure.links
 * @param startId 起点节点 id
 * @returns 所有可达 Module 节点的 id 数组
 */
export function getReachableModules(
  nodes: GraphNode[],
  links: GraphLink[],
  startId: string,
): string[] {
  const nodeMap = new Map(nodes.map((n) => [n.id, n]));
  const linksFrom = new Map<string, GraphLink[]>();
  links.forEach((link) => {
    if (!linksFrom.has(link.source)) linksFrom.set(link.source, []);
    linksFrom.get(link.source)!.push(link);
  });

  // 起点集合 = 自己 + 直接 Structure/Config 关联的 Module
  const startNodes = new Set<string>([startId]);
  const directLinks = linksFrom.get(startId) || [];
  directLinks.forEach((link) => {
    if (link.type === 'Structure' || link.type === 'Config') {
      const targetNode = nodeMap.get(link.target);
      if (targetNode && targetNode.type === 'Module') {
        startNodes.add(link.target);
      }
    }
  });

  // BFS 全边遍历
  const reachableModules = new Set<string>();
  const visited = new Set<string>();
  const queue = Array.from(startNodes);
  while (queue.length > 0) {
    const currentId = queue.shift()!;
    if (visited.has(currentId)) continue;
    visited.add(currentId);
    const node = nodeMap.get(currentId);
    if (!node) continue;
    if (node.type === 'Module') {
      reachableModules.add(currentId);
    }
    const outLinks = linksFrom.get(currentId) || [];
    outLinks.forEach((link) => {
      // analyzer 逻辑：全类型边都遍历
      if (!visited.has(link.target)) {
        queue.push(link.target);
      }
    });
  }
  return Array.from(reachableModules);
}
