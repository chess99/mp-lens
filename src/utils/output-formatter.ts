import chalk from 'chalk';
import * as path from 'path';
import { OutputOptions } from '../types/command-options';

/**
 * 格式化未使用文件的输出
 */
export function formatOutput(
  unusedFiles: string[],
  options: OutputOptions
): string {
  const { format, projectRoot } = options;
  
  if (format === 'json') {
    return formatAsJson(unusedFiles, projectRoot);
  } else {
    return formatAsText(unusedFiles, projectRoot);
  }
}

/**
 * 格式化为JSON输出
 */
function formatAsJson(unusedFiles: string[], projectRoot: string): string {
  const result = {
    unusedFiles: unusedFiles.map(file => ({
      absolutePath: file,
      relativePath: path.relative(projectRoot, file),
      type: path.extname(file).replace('.', '')
    })),
    totalCount: unusedFiles.length,
    timestamp: new Date().toISOString()
  };
  
  return JSON.stringify(result, null, 2);
}

/**
 * 格式化为文本输出
 */
function formatAsText(unusedFiles: string[], projectRoot: string): string {
  if (unusedFiles.length === 0) {
    return chalk.green('未发现未使用的文件。');
  }
  
  // 按照文件类型分类
  const filesByType: Record<string, string[]> = {};
  
  for (const file of unusedFiles) {
    const ext = path.extname(file).replace('.', '') || 'unknown';
    
    if (!filesByType[ext]) {
      filesByType[ext] = [];
    }
    
    filesByType[ext].push(file);
  }
  
  // 构建输出文本
  let output = chalk.yellow(`找到 ${unusedFiles.length} 个未使用的文件:\n`);
  
  // 按照文件类型输出
  for (const [type, files] of Object.entries(filesByType)) {
    output += chalk.cyan(`${type.toUpperCase()} 文件 (${files.length}):\n`);
    
    for (const file of files) {
      // 显示相对路径而非绝对路径
      const relativePath = path.relative(projectRoot, file);
      output += `  ${chalk.white(relativePath)}\n`;
    }
  }
  
  return output;
} 