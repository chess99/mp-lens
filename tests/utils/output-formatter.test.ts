import chalk from 'chalk';
import * as path from 'path';
import { OutputOptions } from '../../src/types/command-options';
import { formatOutput } from '../../src/utils/output-formatter';

// Mock path module
jest.mock('path', () => ({
  ...jest.requireActual('path'), // Keep original methods if needed
  relative: jest.fn((from, to) => to.replace(from + path.sep, '')), // Simple relative path mock
  extname: jest.fn((p) => jest.requireActual('path').extname(p)), // Use actual extname logic initially
}));

// Mock chalk
jest.mock('chalk', () => {
  // Define the mock implementation inside the factory
  const mockChalkInstance = {
    bold: jest.fn((text: string) => `[bold]${text}[/bold]`),
    red: jest.fn((text: string) => `[red]${text}[/red]`),
    yellow: jest.fn((text: string) => `[yellow]${text}[/yellow]`),
    green: jest.fn((text: string) => `[green]${text}[/green]`),
    blue: jest.fn((text: string) => `[blue]${text}[/blue]`),
    gray: jest.fn((text: string) => `[gray]${text}[/gray]`),
    white: jest.fn((text: string) => `[white]${text}[/white]`),
    cyan: jest.fn((text: string) => `[cyan]${text}[/cyan]`),
  };
  // Return the mock instance, potentially wrapped if chalk is a function/class
  // Assuming chalk is an object with methods based on usage
  return mockChalkInstance;
});

describe('Output Formatter', () => {
  const projectRoot = '/path/to/project';
  const unusedFiles = [
    '/path/to/project/src/components/button.wxml',
    '/path/to/project/src/components/button.wxss',
    '/path/to/project/src/pages/index/index.js',
    '/path/to/project/src/utils/unused.ts',
    '/path/to/project/assets/logo.png',
    '/path/to/project/no_extension_file',
  ];
  const emptyUnusedFiles: string[] = [];

  // Reset mocks before each test
  beforeEach(() => {
    jest.clearAllMocks();
    // Configure mocks for specific test needs if necessary
    (path.relative as jest.Mock).mockImplementation((from, to) => {
      // Ensure consistent path separators for snapshot testing
      return to
        .replace(from, '')
        .replace(/^[\/\\]/, '')
        .replace(/\\/g, '/');
    });
    (path.extname as jest.Mock).mockImplementation((p) => jest.requireActual('path').extname(p));
  });

  describe('formatAsJson', () => {
    const options: OutputOptions = { format: 'json', projectRoot };

    it('should return correct JSON format for multiple unused files', () => {
      const output = formatOutput(unusedFiles, options);
      const parsedOutput = JSON.parse(output);

      // Fix timestamp before comparison
      expect(parsedOutput.timestamp).toBeDefined();
      delete parsedOutput.timestamp;

      expect(parsedOutput).toEqual({
        unusedFiles: [
          {
            absolutePath: unusedFiles[0],
            relativePath: 'src/components/button.wxml',
            type: 'wxml',
          },
          {
            absolutePath: unusedFiles[1],
            relativePath: 'src/components/button.wxss',
            type: 'wxss',
          },
          { absolutePath: unusedFiles[2], relativePath: 'src/pages/index/index.js', type: 'js' },
          { absolutePath: unusedFiles[3], relativePath: 'src/utils/unused.ts', type: 'ts' },
          { absolutePath: unusedFiles[4], relativePath: 'assets/logo.png', type: 'png' },
          { absolutePath: unusedFiles[5], relativePath: 'no_extension_file', type: '' }, // No extension
        ],
        totalCount: unusedFiles.length,
      });
    });

    it('should return correct JSON format for zero unused files', () => {
      const output = formatOutput(emptyUnusedFiles, options);
      const parsedOutput = JSON.parse(output);

      // Fix timestamp before comparison
      expect(parsedOutput.timestamp).toBeDefined();
      delete parsedOutput.timestamp;

      expect(parsedOutput).toEqual({
        unusedFiles: [],
        totalCount: 0,
      });
    });

    it('should handle files without extensions correctly in JSON', () => {
      const filesWithNoExt = ['/path/to/project/some_file_no_ext'];
      const output = formatOutput(filesWithNoExt, options);
      const parsedOutput = JSON.parse(output);

      // Fix timestamp before comparison
      expect(parsedOutput.timestamp).toBeDefined();
      delete parsedOutput.timestamp;

      expect(parsedOutput).toEqual({
        unusedFiles: [
          { absolutePath: filesWithNoExt[0], relativePath: 'some_file_no_ext', type: '' },
        ],
        totalCount: 1,
      });
    });
  });

  describe('formatAsText', () => {
    const options: OutputOptions = { format: 'text', projectRoot };

    it('should return correct text format for multiple unused files using snapshot', () => {
      const output = formatOutput(unusedFiles, options);
      // Using snapshot testing for complex text output with mocked colors
      expect(output).toMatchSnapshot();
    });

    it('should return specific message for zero unused files', () => {
      const output = formatOutput(emptyUnusedFiles, options);
      // Exact match for the specific message (including mocked color tags)
      expect(output).toBe('[green]未发现未使用的文件。[/green]');
      expect(chalk.green).toHaveBeenCalledWith('未发现未使用的文件。');
    });

    it('should group files by type correctly in text output', () => {
      const unusedFiles = [
        '/path/to/project/src/components/button.wxml',
        '/path/to/project/src/components/button.wxss',
        '/path/to/project/src/pages/index/index.js',
        '/path/to/project/src/utils/unused.ts',
        '/path/to/project/assets/logo.png',
        '/path/to/project/no_extension_file',
      ];
      const options: OutputOptions = { format: 'text', projectRoot };
      const output = formatOutput(unusedFiles, options);
      // Check if type headers are present, including surrounding newlines from the formatter
      expect(output).toContain('[cyan]WXML 文件 (1):\n');
      expect(output).toContain('[cyan]WXSS 文件 (1):\n');
      expect(output).toContain('[cyan]JS 文件 (1):\n');
      expect(output).toContain('[cyan]TS 文件 (1):\n');
      expect(output).toContain('[cyan]PNG 文件 (1):\n');
      expect(output).toContain('[cyan]UNKNOWN 文件 (1):\n');
    });

    it('should handle files without extensions correctly in text output (as UNKNOWN)', () => {
      const projectRoot = '/path/to/project';
      const options: OutputOptions = { format: 'text', projectRoot };
      const filesWithNoExt = ['/path/to/project/some_file_no_ext'];
      const output = formatOutput(filesWithNoExt, options);
      // Remove the leading \n check
      expect(output).toContain('[cyan]UNKNOWN 文件 (1):\n'); // Adjusted
      expect(output).toContain('[white]some_file_no_ext[/white]');
      expect(output).toMatchSnapshot(); // Keep snapshot check
    });
  });
});
