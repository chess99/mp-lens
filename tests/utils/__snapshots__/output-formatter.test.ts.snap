// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Formatter formatAsText should handle files without extensions correctly in text output (as UNKNOWN) 1`] = `
"[yellow]找到 1 个未使用的文件:
[/yellow][cyan]UNKNOWN 文件 (1):
[/cyan]  [white]some_file_no_ext[/white]
"
`;

exports[`Output Formatter formatAsText should return correct text format for multiple unused files using snapshot 1`] = `
"[yellow]找到 6 个未使用的文件:
[/yellow][cyan]WXML 文件 (1):
[/cyan]  [white]src/components/button.wxml[/white]
[cyan]WXSS 文件 (1):
[/cyan]  [white]src/components/button.wxss[/white]
[cyan]JS 文件 (1):
[/cyan]  [white]src/pages/index/index.js[/white]
[cyan]TS 文件 (1):
[/cyan]  [white]src/utils/unused.ts[/white]
[cyan]PNG 文件 (1):
[/cyan]  [white]assets/logo.png[/white]
[cyan]UNKNOWN 文件 (1):
[/cyan]  [white]no_extension_file[/white]
"
`;
